internal/rukpak/provisioner/helm/helm.go

```diff
@@ -0,0 +1,88 @@
+package helm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+
+	"golang.org/x/sync/errgroup"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/chartutil"
+
+	"github.com/operator-framework/operator-controller/internal/rukpak/util"
+	rukpakv1alpha2 "github.com/operator-framework/rukpak/api/v1alpha2"
+)
+
+const (
+	// ProvisionerID is the unique helm provisioner ID
+	ProvisionerID = "core-rukpak-io-helm"
+)
+
+func HandleBundleDeployment(_ context.Context, fsys fs.FS, bd *rukpakv1alpha2.BundleDeployment) (*chart.Chart, chartutil.Values, error) {
```

Comment:
This bit of code is going to change since we don't have BundleDeployment anymore. Plus looks like this is using helm based bundles (from the provisionerID). We need to port the registryV1 piece of implementation, which would be:
1. Converting registryV1 to plain bundles: the conversion code which is called here: https://github.com/operator-framework/rukpak/blob/882714a00b5d99fdfc1b65992206c4e90edfd696/internal/provisioner/registry/registry.go#L22
2. The logic where we create a chart from the manifest objects: https://github.com/operator-framework/rukpak/blob/882714a00b5d99fdfc1b65992206c4e90edfd696/internal/provisioner/plain/plain.go#L28

This would need some amount of refinement in terms of creating the chart from unpacked bundle contents from the file system. Can we remove this from this PR, and work on this separately along with the helm install logic?

Reply:
Yes, this code was just pulled in, it hasn't been updated. I expect that things will be iterated and removed.